source/base/data_out_base.cc

```diff
@@ -7544,6 +7544,50 @@ DataOutInterface<dim, spacedim>::write_pvtu_record(
 }
 
 
+template <int dim, int spacedim>
+void
+DataOutInterface<dim, spacedim>::write_vtu_with_pvtu_record(
+  const std::string &folder,
+  const std::string &filename_without_extension,
+  const unsigned int counter,
+  const unsigned int n_digits_counter,
+  const MPI_Comm &   mpi_communicator) const
+{
+  const unsigned int rank = Utilities::MPI::this_mpi_process(mpi_communicator);
+  const unsigned int n_ranks =
+    Utilities::MPI::n_mpi_processes(mpi_communicator);
+
+  int n_digits = std::ceil(std::log10(std::fabs(n_ranks)));
+
+  std::string filename = folder + filename_without_extension + "_" +
+                         Utilities::int_to_string(counter, n_digits_counter) +
+                         "." + Utilities::int_to_string(rank, n_digits) +
+                         ".vtu";
+  std::ofstream output(filename.c_str());
+  this->write_vtu(output);
+
+  if (rank == 0)
+    {
+      std::vector<std::string> vector;
+      for (unsigned int i = 0; i < n_ranks; ++i)
+        {
+          std::string filename =
+            filename_without_extension + "_" +
+            Utilities::int_to_string(counter, n_digits_counter) + "." +
+            Utilities::int_to_string(i, n_digits) + ".vtu";
+
+          vector.push_back(filename.c_str());
```

Comment:
```
clang-tidy-6.0 -header-filter=/jenkins/workspace/dea/jenkins/workspace/dealii-tidy_PR-8904/source/base/data_out_base.cc:7579:18: error: use emplace_back instead of push_back [modernize-use-emplace,-warnings-as-errors]
```